code/modules/reagents/chemistry/holder.dm

```diff
@@ -167,13 +168,19 @@
 	if(!glob_reagent)
 		stack_trace("[my_atom] attempted to add a reagent called '[reagent]' which doesn't exist. ([usr])")
 		return FALSE
-
-	var/datum/reagent/D = GLOB.chemical_reagents_list[reagent]
 	if(isnull(added_purity)) //Because purity additions can be 0
-		added_purity = D.creation_purity //Usually 1
-
+		added_purity = glob_reagent.creation_purity //Usually 1
 	if(!added_ph)
-		added_ph = D.ph
+		added_ph = glob_reagent.ph
+
+	//Split up the reagent if it's in a mob
+	var/has_split = FALSE
+	if(is_type_in_list(my_atom, list(/mob/living/carbon, /obj/item/organ/stomach)) && !ignore_splitting) //Stomachs are a pain - they will constantly call on_mob_add unless we split on addition to stomachs, but we also want to make sure we don't double split
```

Comment:
Is this the only way to do this? Cause I really don't like this

Reply:
I could make it a holder flag?

Reply:
I'll make it a holder flag